internal/handlers/pg/pgdb/pool.go

```diff
@@ -69,6 +69,9 @@ func NewPool(ctx context.Context, uri string, logger *zap.Logger, p *state.Provi
 			return lazyerrors.Error(err)
 		}
 
+		// After moving pg to the new architecture, we should stop storing the version in the state,
```

Comment:
No, we should keep storing it in the state because it is used in other places. See references to State's HandlerVersion field

Reply:
Well, it is only needed to send telemetry.

Reply:
So? Does this PR send the SQLite version to telemetry?

Reply:
My comment is that "it is used in other places". There is only one place - telemetry - not "places". I'm ok to store this info twice, I just don't think it's really needed.

Reply:
There is one place where it is stored and two places where it is used.

I agree we should not store that information twice. We also should report in the telemetry, and that code already uses state. So why this PR stores it the in second place?